### DISCUSSION TITLE
Does Serverpod support attaching the auth key in the HttpRequest headers?

### DISCUSSION THREAD

**Discussion by davideroffo** [serverpod/serverpod#657] [Q&A] [answered]
Does Serverpod support attaching the auth key in the HttpRequest headers?

I would like to implement a custom `authenticationHandler` and I want to pass my JWT token in the HttpRequest header instead of passing it in the query params.

**Comment by vlidholt** [comment, marked as answer]
Currently, this is not supported. Can you describe the use case for why you want to do this?

**Reply by davideroffo** [reply]
It's just for security purposes, when using JWT as bearer tokens, it's advisable to avoid sending them in the URL. 

[RFC 6750](https://www.rfc-editor.org/rfc/rfc6750):

> Don't pass bearer tokens in page URLs: Bearer tokens SHOULD NOT be passed in page URLs (for example, as query string parameters).
> 
> Instead, bearer tokens SHOULD be passed in HTTP message headers or message bodies for which confidentiality measures are taken.
> 
> Browsers, web servers, and other software may not adequately secure URLs in the browser history, web server logs, and other data structures. If bearer tokens are passed in page URLs, attackers might be able to steal them from the history data, logs, or other unsecured locations.

In my case, implementing a custom `authenticationHandler` I need to verify (server-side) the validity of JWT tokens, and would like to be compliant with RFC 6750.

**Reply by vlidholt** [reply]
When using SSL the URL is fully encrypted, so it is not a security issue as long as the URL isn't displayed in clear text (e.g. in a browser bar). But we may want to move this to the header in version 2 of Serverpod.

I added an issue we can track here #663
